packages/next/src/shared/lib/runtime-config.external.ts

```diff
@@ -1,9 +1,23 @@
 let runtimeConfig: any
 
+/**
+ * @deprecated Runtime config is deprecated and will be removed in Next.js 16.
+ */
 export default () => {
+  // Don't use picocolors here to isolate the external dependency.
+  console.warn(
+    `Detected the usage of runtime config. Runtime config is deprecated and will be removed in Next.js 16.`
+  )
```

Comment:
You can put a module level variable that checks if we warned already. Doesn't need to use any helpers/abstractions. Just a simple variable to not spam these warnings.